myhoard/controller.py

```diff
@@ -548,60 +542,54 @@ def collect_binlogs_to_purge(
-            file_storage = site_transfers.get(site_name)
+    def get_backups(self) -> Dict[str, Backup]:
+        backups = {}
+        for site_name, site_config in self.backup_sites.items():
+            file_storage = self.site_transfers.get(site_name)
             if file_storage is None:
                 file_storage = get_transfer(site_config["object_storage"])
-                site_transfers[site_name] = file_storage
+                self.site_transfers[site_name] = file_storage
             streams = list(file_storage.list_prefixes(site_name))
             for site_and_stream_id in streams:
-                basebackup_compressed_size = None
-                basebackup_info = {}
+                basebackup_compressed_size: Optional[float] = None
+                basebackup_info: BaseBackup = self.seen_basebackup_infos.get(site_and_stream_id, {})
                 broken_info = {}
                 closed_info = {}
                 completed_info = {}
                 for info in file_storage.list_iter(site_and_stream_id):
                     file_name = info["name"].rsplit("/", 1)[-1]
                     if file_name == "basebackup.xbstream":
                         basebackup_compressed_size = info["size"]
-                    elif file_name == "basebackup.json":
+                    elif file_name == "basebackup.json" and not basebackup_info:
                         # The basebackup info json contents never change after creation so we can use cached
                         # value if available to avoid re-fetching the same content over and over again
-                        basebackup_info = seen_basebackup_infos.get(site_and_stream_id)
-                        if basebackup_info is None:
-                            info_str, _ = file_storage.get_contents_to_string(info["name"])
-                            basebackup_info = json.loads(info_str.decode("utf-8"))
-                            seen_basebackup_infos[site_and_stream_id] = basebackup_info
+                        info_str, _ = file_storage.get_contents_to_string(info["name"])
+                        basebackup_info = json.loads(info_str.decode("utf-8"))
+                        self.seen_basebackup_infos[site_and_stream_id] = basebackup_info
                     elif file_name == "broken.json":
                         broken_info = parse_fs_metadata(info["metadata"])
                     elif file_name == "closed.json":
                         closed_info = parse_fs_metadata(info["metadata"])
                     elif file_name == "completed.json":
                         completed_info = parse_fs_metadata(info["metadata"])
 
-                if basebackup_info and basebackup_compressed_size:
-                    basebackup_info = dict(basebackup_info, compressed_size=basebackup_compressed_size)
+                if basebackup_info and basebackup_compressed_size is not None:
+                    basebackup_info["compressed_size"] = basebackup_compressed_size
```

Comment:
**question** Shouldn't `basebackup.json` already contain the compressed size? This looked quite weird tbh, left it as it is since I am just focusing on typing rather than behavior

Reply:
Looks like it does. I checked the private MyHoard repo with development history and it was not there at one point during development, this is probably a leftover from those times that is not needed because in any officially released version the field is always present.